clustered/server/src/main/java/org/ehcache/clustered/server/ConcurrencyStrategies.java

```diff
@@ -44,17 +46,20 @@ public DefaultConcurrencyStrategy(int bucketCount) {
 
     @Override
     public int concurrencyKey(EntityMessage entityMessage) {
-      if (entityMessage instanceof ConcurrentEntityMessage) {
+      if (entityMessage instanceof ServerStoreOpMessage.GetMessage) {
+        return UNIVERSAL_KEY;
+      } else if (entityMessage instanceof ConcurrentEntityMessage) {
         ConcurrentEntityMessage concurrentEntityMessage = (ConcurrentEntityMessage) entityMessage;
-        return DEFAULT_KEY + Math.abs(concurrentEntityMessage.concurrencyKey() % bucketCount);
+        return DATA_CONCURRENCY_KEY_OFFSET + Math.abs(concurrentEntityMessage.concurrencyKey() % bucketCount);
```

Comment:
Here you must use the exact same logic as in `org.ehcache.clustered.server.offheap.OffHeapServerStore#segmentFor` and since both are inside the server code, I would extract a method that is responsible for doing it.